guides/micronaut-openapi-generator-client/micronaut-openapi-generator-client.adoc

```diff
@@ -0,0 +1,338 @@
+* inside project settings, open application settings,
+* find the `Keys and tokens` tab,
+* inside the tab find the `API key and secret` section and click `Regenerate` to retrieve a new pair of keys.
+====
+
+=== Configure Authorization for Micronaut Client
+
+We will modify the `application.yml` to include the following configuration:
+resource:application.yml[tag=authorization]
+<1> We add a value to the `miconaut.security.oauth2.clients`. We will add it with the key of `twitter`.
+<2> Set the authorization flow to `client_secret_basic`. This authorization was created for applications that need
+to access the resources on behalf of itself and requires an `id` and a `secret` parameter, and a token url.
+<3> Specify the url for retrieving tokens. We found it in the Twitter documentation.
+<4> Specify the `client-id` and `client-secret` parameters. You could directly put your id and secret directly as values
+in the configuration, however this highly discouraged due to security concerns. We will point to environment variables
+instead.
+<5> Specify that this authorization should be used for all client requests (You would probably want to modify this value
+if you wanted to use multiple clients or reach paths that do not require authorization).
+
+Before running the application we will need to set the values of `client-id` and `client-secret` using environment
+variables. To do it, paste your `API key` and `API secret` into the following command and run in terminal:
+[source,bash]
+----
+export TWITTER_AUTH_CLIENT_ID=XXXXXXXXXXXXXXXXXXXXXXXXX
+export TWITTER_AUTH_CLIENT_SECRET=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+----
+
+[WARN]
+Do not forget to add these lines before running any tests or application. If you are using an IDE, there probably should
+be an option to add environment variables in the run settings. View the documentation of your specific IDE for details.
+
+Now, the client should authorize correctly, and we will be able to send requests.
+
+== Testing the Client
+
+You can see that four APIs were generated in the project. These APIs have paths logically distributed between them.
+You can look at link:https://developer.twitter.com/en/docs/twitter-api[Twitter API v2] documentation for description on
+each of the paths, and notice that the sections are also logically split into `Tweets`, `Users` and `Spaces`, as well.
+
+We will show how we can use the Twitter API by writing some simple tests using the generated Micronaut client.
+First, we will use the link:https://developer.twitter.com/en/docs/twitter-api/tweets/counts/introduction[recent tweet counts API]
+to get the number of tweets about Toronto in the last 7 days. Open the `TweetsApiTest` and rewrite the contents of the file with the following:
+
+test:../twitter/api/TweetsApiTest[tags=imports|class-begin|test-1|class-end]
+<1> Annotating the test with `@MicronautTest` will create a Micronaut context and allows using various Micronaut
```

Comment:
The common callout text is: "Annotate the class with `@MicronautTest` so the Micronaut framework will initialize the application context and the embedded server. [More info](https://micronaut-projects.github.io/micronaut-test/latest/guide/)"

I think the description I have makes more sense in this case. There is no need for the embeded server here, and it may confuse the reader.